src/hyperloglog.rs

```diff
@@ -151,7 +152,8 @@ impl<'a, const P: usize, const W: usize> RepresentationTrait for HyperLogLog<'a,
     /// Return memory size of `HyperLogLog`
     #[inline]
     fn size_of(&self) -> usize {
-        size_of::<usize>() + size_of_val(self.data)
+        // The length of the slice, plus the pointer of the slice reference, and the size of the slice itself.
+        size_of::<usize>() + size_of::<usize>() + size_of_val(self.data)
```

Comment:
We technically don't store pointer of the slice reference in memory, as it's stored in "data" field of `CardinalityEstimator` struct.

I'd suggest to measure allocations of `Vec<CardinalityEstimator>` let's say of 1,000 elements of same size.
Also heap allocations can be measured with https://docs.rs/dhat/latest/dhat/ as it's done in https://github.com/cloudflare/cardinality-estimator/blob/main/benches/cardinality_estimator.rs#L148
We could try to compare `mem_dbg` to `size_of` to `dhat` measured allocations.

Reply:
Added more comments regarding this in commit [953cea8](https://github.com/cloudflare/cardinality-estimator/pull/5/commits/953cea89f2fd19162def0e171b6e162443f3fd63). While I am all for comparing the results of `mem-dbg` with those of other crates (I think the authors already have done some of that), in this case I am pretty sure about the correctness of the estimation of the size of a slice (len (usize) + pointer (usize) + size of values therein)